File: test/language/templates/cpp/InstantiateOnlyNestedTest.cpp
#include "gtest/gtest.h"

#include "templates/instantiate_only_nested/InstantiateOnlyNested.h"

namespace templates
{
namespace instantiate_only_nested
{

TEST(InstantiateOnlyNestedTest, readWrite)
{
    InstantiateOnlyNested instantiateOnlyNested;
    instantiateOnlyNested.setTest32(pkg::Test_uint32{N32{13}});

    zserio::BitBuffer bitBuffer = zserio::BitBuffer(1024 * 8);
    zserio::BitStreamWriter writer(bitBuffer);
    instantiateOnlyNested.write(writer);

    zserio::BitStreamReader reader(writer.getWriteBuffer(), writer.getBitPosition(), zserio::BitsTag());
    InstantiateOnlyNested readInstantiateOnlyNested(reader);

    ASSERT_TRUE(instantiateOnlyNested == readInstantiateOnlyNested);
}

} // namespace instantiate_only_nested
} // namespace templates

File: test/language/templates/python/InstantiateOnlyNestedTest.py
import zserio

import Templates

class InstantiateOnlyNestedTest(Templates.TestCase):
    def testReadWrite(self):
        instantiateOnlyNested = self.api.InstantiateOnlyNested(self.api.pkg.Test_uint32(self.api.N32(13)))

        writer = zserio.BitStreamWriter()
        instantiateOnlyNested.write(writer)
        reader = zserio.BitStreamReader(writer.byte_array, writer.bitposition)
        readInstantiateOnlyNested = self.api.InstantiateOnlyNested()
        readInstantiateOnlyNested.read(reader)
        self.assertEqual(instantiateOnlyNested, readInstantiateOnlyNested)
